src/plugins/intel_gpu/tests/test_cases/range_gpu_test.cpp

```diff
@@ -29,18 +33,19 @@ struct RangeArgs {
     RangeArg start { dt, "start" };
     RangeArg stop { dt, "stop" };
     RangeArg step { dt, "step" };
-    explicit RangeArgs(data_types dt) :
-        dt { dt } {
-    }
-    memory::ptr run(int outLen) const {
+    explicit RangeArgs(data_types dt) : dt { dt } {}
+
+    memory::ptr run(int outLen, bool use_new_shape_infer) const {
         topology topology;
         start.addTo(topology);
         stop.addTo(topology);
         step.addTo(topology);
-        topology.add(range { "range", { start.name, stop.name, step.name }, { dt, format::bfyx,
-            tensor { spatial(outLen) } } });
+        topology.add(range { "range", { start.name, stop.name, step.name }, { dt, format::bfyx, tensor{batch(outLen)} } });
+
+        build_options bo;
+        bo.set_option(build_option::allow_new_shape_infer(use_new_shape_infer));
 
-        network network { tests::get_test_engine(), topology };
+        network network { tests::get_test_engine(), topology, bo };
```

Comment:
This change allows testing with only new shape_infer, though we still need to use legacy shape infer for static shape.

Reply:
No, this test support all (new shape infer and legacy shape infer). test params set the use_new_shape_infer flags. if you check range_test_param_generator, you can find the code to set allow_new_shape_infer flags.

Reply:
Oh I see. Thanks.